### PR TITLE
RFQ: take transactor.Value into account in gas middleware

### PR DESCRIPTION
**Pull request by dwasse**
## Summary by CodeRabbit

- **New Features**
  - Enhanced gas sufficiency checks across different blockchain networks to ensure transactions are processed only when sufficient gas is available on both the origin and destination chains.

- **Bug Fixes**
  - Improved error handling and conditions in gas sufficiency checks to prevent transaction failures.

- **Tests**
  - Updated tests to cover new scenarios for gas sufficiency, including edge cases with `nil` and non-zero gas values.

### CODE REVIEW

File: services/rfq/relayer/service/statushandler.go [hunk 1]
```diff
@@ -107,12 +107,33 @@ func (r *Relayer) deadlineMiddleware(next func(ctx context.Context, span trace.S
...
+			span.SetAttributes(attribute.String("dest_gas_value", destGasValue.String()))
+		}
+		sufficientGasDest, err = r.inventory.HasSufficientGas(ctx, int(req.Transaction.DestChainId), destGasValue)
 		if err != nil {
-			return fmt.Errorf("could not check gas: %w", err)
+			return fmt.Errorf("could not check gas on dest: %w", err)
 		}
-		if !sufficientGas {
+
+		if !sufficientGasOrigin || sufficientGasDest {
```
[the remark this answers is not in the document]
Comment (ChiTimesChi) [reply services/rfq/relayer/service/statushandler.go:136]: @dwasse ^
